File: OfflineWeb/OfflineWebFE/library/Doc.php
<?php

/**
 * Description of Doc
 *
 * @author papa2
 */

require_once 'Utility.php';

class Doc {
    
    private static $logger = NULL;
    
    private $title;
    private $docId;
    private $docAbstarct;
    private $type;
    private $source;
    
    function __construct($title="", $type="", $source="", $docId="", $docAbstarct="") {
        self::$logger = Utility::getLogger();
        $this->title = $title;
        $this->type = $type;
        $this->source = $source;
        $this->docAbstarct = $docAbstarct;
        $this->docId = $docId;
    }
    
    function getTitle() {
        return $this->title;
    }

    function getType() {
        return $this->type;
    }

    function getSource() {
        return $this->source;
    }
    
    function getDocId() {
        return $this->docId;
    }

    function getDocAbstarct() {
        if ($this->docAbstarct === NULL || $this->docAbstarct === "") {
            return "";
        } else {
            return $this->docAbstarct . "...";
        }
    }

    function setTitle($title) {
        $this->title = $title;
    }

    function setType($type) {
        $this->type = $type;
    }

    function setSource($source) {
        $this->source = $source;
    }
    
    function setDocId($docId) {
        $this->docId = $docId;
    }

    function setDocAbstarct($docAbstarct) {
        $this->docAbstarct = $docAbstarct;
    }

       
    function hasThumbnail() {
        if ($this->source === "youtube") {
            return TRUE;
        } else {
            return FALSE;
        }
    }

    function getThumbnailURL() {
        $thumbURL = Utility::getConfigProperty('youtubeImageURL') . "/" . $this->docId . ".jpg";
        self::$logger->info("getThumbnailURL : " . $thumbURL);
        return $thumbURL;
    }
}

File: OfflineWeb/OfflineWebBE/src/java/offlineweb/api/bean/Doc.java

package offlineweb.api.bean;

import javax.xml.bind.annotation.XmlRootElement;

/**
 *
 * @author papa2
 */

@XmlRootElement
public class Doc {

    private String title;
    private String docId;
    private String docAbstract;
    private String type;
    private String source;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDocId() {
        return docId;
    }

    public void setDocId(String docId) {
        this.docId = docId;
    }

    public String getDocAbstract() {
        return docAbstract;
    }

    public void setDocAbstract(String docAbstract) {
        this.docAbstract = docAbstract;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    @Override
    public String toString() {
        return "Doc{" + "title=" + title + ", docId=" + docId + ", source=" + source + '}';
    }

}
